src/bibx/_entities/collection.py

```diff
@@ -38,6 +39,27 @@ def citation_pairs(self) -> Iterable[Tuple[Article, Article]]:
                 else:
                     yield article, reference
 
+    def count_by_year(self) -> Dict[int, int]:
+        """
+        Returns a dictionary where the key is the year of publication and the value is the number of articles
+        published that year. The dictionary starts from the oldest article to the current year consecutively.
+        If a year has no publications the value will be zero.
+
+        :return: a dictionary with the number of articles published each year.
+        """
+        current_year = datetime.date.today().year
+        primary_year = min([article.year for article in self.articles])
+        years = {}
+        for year in range(primary_year, current_year + 1):
+            years[year] = 0
+
+        for article in self.articles:
+            if article.year in years:
```

Comment:
We need to handle the case where `article.year` is `None`. If that's the case we just ignore the article.

Reply:
:+1: